site/app/controllers/admin/ReportController.php

```diff
@@ -37,8 +37,52 @@ public function showReportPage() {
         $this->core->getOutput()->renderOutput(array('admin', 'Report'), 'showReportUpdates');
     }
     
-    public function showCSVReport() {
+    public function generateCSVReport() {
+        $students = $this->core->getQueries()->getAllUsers();
+        $student_ids = array_map(function($stu) {return $stu->getId();}, $students);
+        $gradeables = $this->core->getQueries()->getGradeables(null, $student_ids);
+        $results = array();
+        $results['header_model'] = array('First' => 'First Name', 'Last'=> 'Last Name', 'reg_section' => 'Registration Section');
+        foreach($gradeables as $gradeable) {
+            $student_id = $gradeable->getUser()->getId();
+            if(!isset($results[$student_id])) {
+                $results[$student_id] = array('First'=>$gradeable->getUser()->getDisplayedFirstName(), 'Last' => $gradeable->getUser()->getLastName(), 'reg_section' => $gradeable->getUser()->getRegistrationSection());
+            }
+            $g_id = $gradeable->getId();
+            if(!isset($results['header_model'][$g_id])) {
+                $results['header_model'][$g_id] = $g_id.": ".($gradeable->getTotalAutograderNonExtraCreditPoints() + $gradeable->getTotalTANonExtraCreditPoints());
+            }
+            $results[$student_id][$g_id] = $gradeable->getGradedTAPoints() + $gradeable->getGradedAutograderPoints();
+        }
+        
+        $nl = "\n";
+        $csv_output = "";
+        foreach($results as $id => $student) {
+            $student_line = array();
+            if($id === 'header_model') {
+                $student_line[] = "UserId";
+            }
+            else {
+                $student_line[] = $id;
+            }
+            $student_line[] = $student['First'];
+            $student_line[] = $student['Last'];
+            $student_line[] = $student['reg_section'];
+            foreach($results['header_model'] as $grade_id => $grade) {
+                if($grade_id == 'First' || $grade_id == 'Last' || $grade_id == 'reg_section') {
+                    continue;
+                }
+                $student_line[] = $student[$grade_id];
+            }
+            $csv_output .= implode(",",$student_line).$nl;
+        }
         
+        header("Content-Type: text/plain");
+        header('Content-Disposition: attachment; filename=hwserver-report.csv');
+        header("Content-Length: " . strlen($csv_output));
+        echo $csv_output;
+        exit;
```

Comment:
Eh, ignore the `$csv_output` return for now and that can be done when we actually add tests for this function.

Change [this line](https://github.com/Submitty/Submitty/blob/master/site/public/index.php#L212) to be instead `$core->getOutput()->displayOutput()`. Then add what I mentioned above within the Output library.